lib/portage/util/locale.py

```diff
@@ -116,29 +139,21 @@ def check_locale(silent=False, env=None):
         except KeyError:
             pass
 
-    pid = os.fork()
-    if pid == 0:
-        try:
-            if env is not None:
-                try:
-                    locale.setlocale(locale.LC_CTYPE, portage._native_string(mylocale))
-                except locale.Error:
-                    os._exit(2)
-
-            ret = _check_locale(silent)
-            if ret is None:
-                os._exit(2)
-            else:
-                os._exit(0 if ret else 1)
-        except Exception:
-            traceback.print_exc()
-            os._exit(2)
-
-    pid2, ret = os.waitpid(pid, 0)
-    assert pid == pid2
+    # TODO: Make async version of check_locale and call it from
+    # EbuildPhase instead of config.environ(), since it's bad to
+    # synchronously wait for the process in the main event loop
+    # thread where config.environ() tends to be called.
+    proc = multiprocessing.Process(
+        target=_set_and_check_locale,
+        args=(silent, env, None if env is None else portage._native_string(mylocale)),
+    )
+    proc.start()
+    proc = portage.process.MultiprocessingProcess(proc)
+    await proc.wait()
+
     pyret = None
-    if os.WIFEXITED(ret):
```

Comment:
At some point, we should check over our instances of WIFEXITED and other shenanigans to see if we can port to ` os.waitstatus_to_exitcode(status)`.

Reply:
Yeah, I didn't realize it was that prevalent but now I see there 7 remaining.